torch/csrc/jit/passes/alias_analysis.cpp

```diff
@@ -15,7 +15,8 @@ namespace jit {
 // introduce more granularity here (e.g. List[int] will never alias
 // List[float]).
 c10::optional<TypeKind> AliasDb::getMutableTypeKind(const TypePtr& type) {
-  if (type->isSubtypeOf(TensorType::get())) {
+
+  if (type->cast<TensorType>()) {
```

Comment:
you can just add case TypeKind::TensorType to the statements below. This looks like a holdover from when there were multiple tensor types.